tlvc/arch/i86/drivers/block/ll_rw_blk.c

```diff
@@ -154,21 +139,28 @@ static void add_request(struct blk_dev_struct *dev, struct request *req)
 {
     register struct request *tmp;
 
-    clr_irq();
+    //clr_irq();
```

Comment:
Not sure why disabling interrupts was removed here: this introduces a big-time race condition, where the following line `tmp = dev->current_request` is checked for NULL, and `end_request` called at interrupt time from the direct floppy driver executes `CURRENT = req->rq_next` in blk.h which writes `dev->current_request`.

When the race condition causes an error, what happens is either the next pending I/O request in the queue or the next request to be added in `make_request` is dropped or the request queue becomes corrupted.

Reply:
Thanks @ghaerr - great catch!!!
It was removed because `add_request()` was called with interrupts disabled for an experimental period, then never fixed.